dev/release/publish-to-maven.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+###
+# This file is based on the script release-build.sh in Spark
+###
+
+function usage {
+  local NAME=$(basename $0)
+  cat << EOF
+usage: $NAME options
+
+Publish signed artifacts to Maven.
+
+Options
+  -u ASF_USERNAME - Username of ASF committer account
+  -r LOCAL_REPO - path to temporary local maven repo (created and written to by 'build-release-comet.sh')
+
+  One of -p or -v must be specified
+
+The following will be prompted for -
+  ASF_PASSWORD - Password of ASF committer account
+  GPG_KEY - GPG key used to sign release artifacts
+  GPG_PASSPHRASE - Passphrase for GPG key
+EOF
+  exit 1
+}
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+COMET_HOME_DIR=$SCRIPT_DIR/../..
+
+ASF_USERNAME=""
+ASF_PASSWORD=""
+
+NEXUS_ROOT=https://repository.apache.org/service/local/staging
+NEXUS_PROFILE=789e15c00fd47
+
+while getopts "u:r:h" opt; do
+  case $opt in
+    u) ASF_USERNAME="$OPTARG" ;;
+    r) LOCAL_REPO="$OPTARG" ;;
+    h) usage ;;
+    \?) error "Invalid option. Run with -h for help." ;;
+  esac
+done
+
+if [ "$LOCAL_REPO" == "" ]
+then
+  "Please provide the local Maven repository (from 'build-release-comet.sh')"
+  usage
+fi
+
+if [ "$ASF_USERNAME" == "" ]
+then
+  read -p "ASF Username : " ASF_USERNAME && echo ""
+fi
+# Read some secret information
+read -s -p "ASF Password : " ASF_PASSWORD && echo ""
+read -s -p "GPG Key (Optional): " GPG_KEY && echo ""
+read -s -p "GPG Passphrase : " GPG_PASSPHRASE && echo ""
+
+if [ "$ASF_USERNAME" == "" ] || [ "$ASF_PASSWORD" == "" ] || [ "$GPG_PASSPHRASE" = "" ]
+then
+  echo "Missing credentials"
+  exit 1
+fi
+
+# Default GPG command to use
+GPG="gpg"
```

Comment:
The Spark version of this script has `GPG="gpg --no-tty --batch --pinentry-mode loopback"` and I wonder if we should use any of those flags?

Reply:
I think this was needed in Spark to allow it to run unattended in a docker container. I didn't run into any issues without these flags.